homeassistant/components/bzutech/config_flow.py

```diff
@@ -0,0 +1,264 @@
+            return await self.async_step_configend()
+        return await self.async_step_option1(user_input)
+
+    async def async_step_option2(self, user_input: dict) -> FlowResult:
+        """Unification of the device configuration menu options."""
+        tempStorage = self.hass.data[DOMAIN]
+        if tempStorage["flowstep"] == 1:
+            tempStorage["selecteddisp"] = 1
+            return await self.async_step_portselect(user_input)
+        if tempStorage["flowstep"] == 2:
+            tempStorage["selectedport"] = 2
+            return await self.async_step_sensorselect(user_input)
+        if tempStorage["flowstep"] == 3:
+            tempStorage["selectedsensor"] = 1
+            return await self.async_step_configend()
+        return await self.async_step_option2(user_input)
+
+    async def async_step_option3(self, user_input: dict) -> FlowResult:
+        """Unification of the device configuration menu options."""
+        tempStorage = self.hass.data[DOMAIN]
+        if tempStorage["flowstep"] == 1:
+            tempStorage["selecteddisp"] = 2
+            return await self.async_step_portselect(user_input)
+        if tempStorage["flowstep"] == 2:
+            tempStorage["selectedport"] = 3
+            return await self.async_step_sensorselect(user_input)
+        if tempStorage["flowstep"] == 3:
+            tempStorage["selectedsensor"] = 2
+            return await self.async_step_configend()
+        return await self.async_step_option3(user_input)
+
+    async def async_step_option4(self, user_input: dict) -> FlowResult:
+        """Unification of the device configuration menu options."""
+        tempStorage = self.hass.data[DOMAIN]
+        if tempStorage["flowstep"] == 1:
+            tempStorage["selecteddisp"] = 3
+            return await self.async_step_portselect(user_input)
+        if tempStorage["flowstep"] == 2:
+            tempStorage["selectedport"] = 4
+            return await self.async_step_sensorselect(user_input)
+        if tempStorage["flowstep"] == 3:
+            tempStorage["selectedsensor"] = 3
+            return await self.async_step_configend()
+        return await self.async_step_option4(user_input)
+
+    async def async_step_option5(self, user_input: dict) -> FlowResult:
+        """Unification of the device configuration menu options."""
+        tempStorage = self.hass.data[DOMAIN]
+        if tempStorage["flowstep"] == 1:
+            tempStorage["disppage"] = tempStorage["disppage"] + 1
+            return await self.async_step_dispselect(user_input)
+        if tempStorage["flowstep"] == 2:
+            tempStorage["selectedport"] = 5
+            return await self.async_step_sensorselect(user_input)
+        if tempStorage["flowstep"] == 3:
+            tempStorage["sensorpage"] = tempStorage["sensorpage"] + 1
+            return await self.async_step_sensorselect(user_input)
+        return await self.async_step_option5(user_input)
```

Comment:
what are you trying to achieve here?

Reply:
i though of it as a way to configure the integration using menus instead of forms, the async_step_optionN handles the selection, is there a more efficient way of doing it? heres how it is running right now

![image](https://i.ibb.co/QMkyyRZ/Captura-de-tela-2024-01-11-154508.png)

Reply:
Can you maybe give some more context on what this code is doing and what kind of purpose it serves? What kind of devices are you connecting with this integration?

Reply:
> Can you maybe give some more context on what this code is doing and what kind of purpose it serves? What kind of devices are you connecting with this integration?

to get data from the API and register a new entity on HA the integration needs 4 informations, login data, Device Chip Id, Device port and Sensor on that port, so the config flow follow the steps: Ask for Login data, authenticate it through the API, gets device/chipids associated with that login data, show them in menu form for the user to select, stores the selected option in a configflow variable by clicking on one of the optionN menuoption, lists the available ports in that device (usually 4) and stores the selected port, then gets from the API the sensors on that DeviceChipid and that Port, lists them and stores it the same way as the Device and Port.

Reply:
Why can't we register all ports automatically?

Reply:
> Why can't we register all ports automatically?

When developing i did not like how it felt when displaying every available sensor from a device at once, selecting the specific port and listing its sensors was more efficient and cleaner.

Reply:
Can you categorize them? Maybe disabled the less used/interesting ones by default?

Reply:
i am working on registering every sensor on a port of a device, so, instead of selecting one sensor of one port of a device, you select one port of a device and every sensor gets registered, do you think this approach would make it better?